metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java

```diff
@@ -244,6 +253,35 @@ protected T constructEntity(Node node) {
             throw new MetadataRepositoryException("Unable to findAll for Type : " + getNodeType(getJcrEntityClass()), e);
         }
     }
+    
+    public int findCount() {
+        return findCount(null);
+    }
+
+    public int findCount(String filter) {
+        StringBuilder bldr = startBaseQuery("[mode:id]");
+        int count = 0;
+        
+        appendJoins(bldr, filter);
+        appendFilter(bldr, filter);
+        
+        try {
+            QueryResult result = JcrQueryUtil.query(getSession(), bldr.toString());
+            
+            if (result != null) {
+                RowIterator rowItr = result.getRows();
+                
+                while (rowItr.hasNext()) {
+                    rowItr.nextRow();
+                    count++;
+                }
+            }
+            
+            return count;
+        } catch (RepositoryException e) {
+            throw new MetadataRepositoryException("Unable to findAll for Type : " + getNodeType(getJcrEntityClass()), e);
+        }
+    }
 
     public <D, R> Iterable<D> findIterable(String query, Class<D> domainClass, Class<R> resultClass) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "domainClass". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)